libs/utils/trace.py

```diff
@@ -888,6 +888,95 @@ def addEventsDeltas(self, df, col_name='len'):
         # Set duration to trace_end - last_event
         df.loc[df.index[-1], 'len'] = self.start_time + self.time_range - df.index[-1]
 
+    def slice_df(self, df, start, end, len_col='len'):
```

Comment:
I would say the API looks more generic if we:
1. rename the method to `trim_df`... since that's doing much more then a "standard slice"
2. replace `len_col` with `column`, which more specifically defines that the values that will be trimmed according to the new start,end boundaries.

The documentation below should better state the expected DF input format, i.e. the column should report deltas among consecutive DFs.

Reply:
Speaking of names, I was wondering if `squash_df` wouldn't be actually clearer. It's not a "clean cut" as would imply slice or trim - you actually squash stuff within an interval.